Refresh.GameServer/Endpoints/Game/Playlists/PlaylistEndpoints.cs

```diff
@@ -0,0 +1,338 @@
+
+            if (level == null)
+                return null;
+            
+            playlists = author == null ? 
+                dataContext.Database.GetPlaylistsContainingLevel(level).ToList() : 
+                dataContext.Database.GetPlaylistsByAuthorContainingLevel(author, level).ToList();
+        }
+        
+        int total = playlists.Count;
+        
+        (int skip, int count) = context.GetPageData();
+
+        // Return the serialized playlists 
+        return new SerializedMinimalLevelList(
+            GameMinimalLevelResponse.FromOldList(playlists.Skip(skip).Take(count), dataContext), 
+            total, 
+            skip
+        );
+    }
+
+    [GameEndpoint("setPlaylistMetaData/{id}", HttpMethods.Post, ContentType.Xml)]
+    [RequireEmailVerified]
+    public Response UpdatePlaylistMetadata(RequestContext context, GameDatabaseContext database, GameUser user, int id, SerializedPlaylist body)
+    {
+        GamePlaylist? playlist = database.GetPlaylistById(id);
+
+        // Handle a bad playlist ID
+        if (playlist == null)
+            return NotFound;
+
+        // Dont allow the wrong user to update playlists
+        if (playlist.Creator.UserId != user.UserId)
+            return Unauthorized;
+        
+        database.UpdatePlaylist(playlist, body);
+        
+        return OK;
+    }
+
+    [GameEndpoint("deletePlaylist/{id}", HttpMethods.Post)]
+    [RequireEmailVerified]
+    public Response DeletePlaylist(RequestContext context, GameDatabaseContext database, GameUser user, int id)
+    {
+        GamePlaylist? playlist = database.GetPlaylistById(id);
+
+        // Handle a bad playlist ID
+        if (playlist == null)
+            return NotFound;
+
+        // Dont allow the wrong user to delete playlists
+        if (playlist.Creator.UserId != user.UserId)
+            return Unauthorized;
+
+        database.DeletePlaylist(playlist);
+            
+        return OK;
+    }
+
+    [GameEndpoint("addToPlaylist/{playlistId}", HttpMethods.Post)]
+    [RequireEmailVerified]
+    public Response AddSlotToPlaylist(RequestContext context, GameDatabaseContext database, GameUser user, int playlistId)
+    {
+        // Extract the slot type, ensuring its set
+        string? slotType = context.QueryString["slot_type"];
+        if (slotType == null)
+            return BadRequest;
+        
+        // Extract the slot ID ensuring its valid
+        if (!int.TryParse(context.QueryString["slot_id"], out int slotId))
+            return BadRequest;
+
+        GamePlaylist? parentPlaylist = database.GetPlaylistById(playlistId);
+
+        // If the parent doesn't exist, exit gracefully
+        if (parentPlaylist == null)
+            return NotFound;
+
+        // Dont let people add slots to other's playlists
+        if (parentPlaylist.Creator.UserId != user.UserId)
+            return Unauthorized;
+
+        // Special handling for playlists
```

Comment:
And comments like this need more clarification, e.g.:

```suggestion
        // Adding a playlist to a playlist requires a special case, since we use `SubPlaylistRelation` internally to record child playlists.
```